src/components/structures/login/Registration.js

```diff
@@ -227,6 +227,9 @@ module.exports = React.createClass({
                     const teamToken = data.team_token;
                     // Store for use /w welcome pages
                     window.localStorage.setItem('mx_team_token', teamToken);
+                    // Set the team token and view homepage
+                    window.matrixChat._teamToken = teamToken;
+                    window.mxDispatcher.dispatch({action: 'view_home_page'});
```

Comment:
Ah, it's apparently done through `onLoggedIn`, which is called before. So I don't think it'll race. Using `notifyNewScreen` with `home` could be better actually.